lib/S3Plugin.js

```diff
@@ -361,6 +412,133 @@ class S3Plugin {
     return stringParam;
   }
 
+  /**
+   * validate given string to match
+   * AWS S3 bucket name format prerequisites
+   */
+    _validateBucketName (name, disableDotsInName=false) {
+      const bucketNameRegex = disableDotsInName ? 
+        // avoiding dots is needed for S3 Transfer Acceleration
+        /(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$/
+        : /(?!(^((2(5[0-5]|[0-4][0-9])|[01]?[0-9]{1,2})\.){3}(2(5[0-5]|[0-4][0-9])|[01]?[0-9]{1,2})$|^xn--|.+-s3alias$))^[a-z0-9][a-z0-9.-]{1,61}[a-z0-9]$/
+      
+        const sanitizedName = bucketNameRegex.test(name)
+  
+      if (sanitizedName === null) {
+        throw new this.context.errors.BadRequestError(`Invalid bucket name format: ${name}`)
```

Comment:
You should return false here instead of throwing because you are already throwing an exception in the action handler l.472 if the name is not valid.